src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+            Span<byte> accData = stackalloc byte[2] { 0, 0 };
+            // Read factory accelerometer trim values
+            ReadByteArray(Register.XA_OFFSET_H, accData);
+            accel_bias_reg.X = BinaryPrimitives.ReadUInt16BigEndian(accData);
+            ReadByteArray(Register.YA_OFFSET_H, accData);
+            accel_bias_reg.Y = BinaryPrimitives.ReadUInt16BigEndian(accData);
+            ReadByteArray(Register.ZA_OFFSET_H, accData);
+            accel_bias_reg.Z = BinaryPrimitives.ReadUInt16BigEndian(accData);
+
+            // Define mask for temperature compensation bit 0 of lower byte of
+            // accelerometer bias registers
+
+            uint mask = 0x01;
+            // Define array to hold mask bit for each accelerometer bias axis
+            byte[] mask_bit = new byte[3];
+
+            // If temperature compensation bit is set, record that fact in mask_bit
+            mask_bit[0] = (((uint)accel_bias_reg.X & mask) == mask) ? (byte)0x01 : (byte)0x00;
+            mask_bit[1] = (((uint)accel_bias_reg.Y & mask) == mask) ? (byte)0x01 : (byte)0x00;
+            mask_bit[2] = (((uint)accel_bias_reg.Z & mask) == mask) ? (byte)0x01 : (byte)0x00;
+
+            // Construct total accelerometer bias, including calculated average
+            // accelerometer bias from above
+            // Subtract calculated averaged accelerometer bias scaled to 2048 LSB/g
+            // (16 g full scale) and keep the mask
+            accel_bias_reg -= acceBias / 8;
+            // Add the "reserved" mask as it was
+            rawData[0] = (byte)(((int)accel_bias_reg.X >> 8) & 0xFF);
+            rawData[1] = (byte)(((int)accel_bias_reg.X & 0xFF) | mask_bit[0]);
+            rawData[2] = (byte)(((int)accel_bias_reg.Y >> 8) & 0xFF);
+            rawData[3] = (byte)(((int)accel_bias_reg.Y & 0xFF) | mask_bit[1]);
+            rawData[4] = (byte)(((int)accel_bias_reg.Z >> 8) & 0xFF);
+            rawData[5] = (byte)(((int)accel_bias_reg.Z & 0xFF) | mask_bit[2]);
+            // Push accelerometer biases to hardware registers
+            WriteRegister(Register.XA_OFFSET_H, rawData[0]);
+            WriteRegister(Register.XA_OFFSET_L, rawData[1]);
+            WriteRegister(Register.YA_OFFSET_H, rawData[2]);
+            WriteRegister(Register.YA_OFFSET_L, rawData[3]);
+            WriteRegister(Register.ZA_OFFSET_H, rawData[4]);
+            WriteRegister(Register.ZA_OFFSET_L, rawData[5]);
+
+            // Restore the previous modes
+            WriteRegister(Register.USER_CTRL, (byte)(userControls | (byte)UserControls.I2C_MST_EN));
+            i2cMaster = (byte)(i2cMaster & (~(byte)(I2cBussFrequency.Freqency348kHz) | (byte)I2cBussFrequency.Freqency400kHz));
+            WriteRegister(Register.I2C_MST_CTRL, i2cMaster);
+            DelayHelper.DelayMilliseconds(10, false);
+
+            // Setup I2C for the AK8963
+            //WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_EN);
```

Comment:
Why are these two writes commented out?

Reply:
Just forgot to remove them. I decided to take out the magnetometer calibration and have it separately. So removing the lines.